code/game/objects/items/weapons/melee/melee_misc.dm

```diff
@@ -45,6 +45,388 @@
+/obj/item/melee/breach_cleaver/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/two_handed, force_wielded = force_wield, force_unwielded = force, icon_wielded = "[base_icon_state]1", wield_callback = CALLBACK(src, PROC_REF(wield)), unwield_callback = CALLBACK(src, PROC_REF(unwield)))
+	AddComponent(/datum/component/parry, _stamina_constant = 2, _stamina_coefficient = 0.5, _parryable_attack_types = NON_PROJECTILE_ATTACKS)
+
+/obj/item/melee/breach_cleaver/examine(mob/user)
+	. = ..()
+	if(isAntag(user))
+		. += "<span class='warning'>When wielded, this blade has different effects depending on your intent, similar to a martial art. \
+			Help intent will strike with the flat, dealing stamina, disarm intent forces them away, grab intent knocks down the target, \
+			and harm intent deals heavy damage.</span>"
+
+/obj/item/melee/breach_cleaver/examine_more(mob/user)
+	. = ..()
+	. += "Massive, heavy, and utterly impractical. This sharpened chunk of steel is too big and too heavy to be called a sword."
+	. += ""
+	. += "The Unathi Breach Cleaver is a weapon the scaled, warlike race favours for its impressive weight and myriad combat applications. \
+	The pinnacle of Moghes' combat technology, it combines all of this knowledge into a massive, heavy slab of alloyed metal that most \
+	species find difficult to lift, let alone use in any sort of fight."
+	. += ""
+	. += "Actually a little lightweight for its size, a Breach Cleaver is unmatched in combat utility as a weapon, a tool for getting into\
+	places and as a slab of armour for the wielder. The leather of the Kar'oche beast, a predator native to Moghes, binds the hilt, \
+	allowing it to be gripped securely by its warrior. The wide blade is often etched with scenes depicting military victories or great hunts."
+
+/obj/item/melee/breach_cleaver/update_icon_state()
+	icon_state = "[base_icon_state]0"
+
+/obj/item/melee/breach_cleaver/proc/wield(obj/item/source, mob/user)
+	to_chat(user, "<span class='notice'>You heave [src] up in both hands.</span>")
+	var/mob/living/carbon/human/H = user
+	H.apply_status_effect(STATUS_EFFECT_BREACH_AND_CLEAVE)
+	update_icon_state()
+
+/obj/item/melee/breach_cleaver/proc/unwield(obj/item/source, mob/user)
+	var/mob/living/carbon/human/H = user
+	H.remove_status_effect(STATUS_EFFECT_BREACH_AND_CLEAVE)
+	update_icon_state()
+
+/obj/item/melee/breach_cleaver/attack_obj(obj/O, mob/living/user, params)
+	if(!HAS_TRAIT(src, TRAIT_WIELDED)) // Only works good when wielded
+		. = ..()
+	else if(ismachinery(O) || isstructure(O)) // This sword hates doors
+		if(SEND_SIGNAL(src, COMSIG_ITEM_ATTACK_OBJ, O, user) & COMPONENT_NO_ATTACK_OBJ)
+			return
+		if(flags & (NOBLUDGEON))
+			return
+		user.changeNext_move(CLICK_CD_MELEE)
+		user.do_attack_animation(O)
+		user.visible_message("<span class='danger'>[user] has hit [O] with [src]!</span>", "<span class='danger'>You hit [O] with [src]!</span>")
+		var/damage = force_wield
+		var/mob/living/carbon/human/H = user
```

Comment:
for consistency, I would typecast this at the beginning of these lines of code instead of relying on `user` for half

Reply:
why was this marked as resolved?

Reply:
In one of the recent pushes, I addressed the issue contained by changing when it was typecasted and how it handled it.